src/big_query/big_query.py

```diff
@@ -72,7 +72,16 @@ def list_table_ids(self, project_id, dataset_id):
             projectId=project_id, datasetId=dataset_id
         )
         while request is not None:
-            tables = request.execute()
+            try:
+                tables = request.execute()
+            except HttpError as ex:
+                if ex.resp.status == 404:
+                    logging.info("Dataset not '%s:%s' Not Found", project_id,
+                               dataset_id)
+                    raise StopIteration
```

Comment:
Is it recommended way in python? Couldn't we just give 'return' here?

Reply:
Return is better. Done.